e2e/loadtest/load_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loadtest
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"runtime"
+	"testing"
+	"time"
+
+	"fortio.org/fortio/fhttp"
+	"fortio.org/fortio/fnet"
+	"github.com/stretchr/testify/require"
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/font"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/plotutil"
+	"gonum.org/v1/plot/vg"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/e2e/testutil"
+)
+
+const (
+	targetNintyninthPercentileLatencyLimit = 200 // ms
+	targetSuccessRate                      = 0.7
+)
+
+var (
+	qps           = flag.Float64("qps", 0, "QPS to run load test")
+	duration      = flag.Int("duration", 10, "Duration of test in seconds")
+	payloadSize   = flag.Int("payloadSize", 256, "Payload size in kilo bytes")
+	targetExample = flag.String("targetExample", "http_headers", "Target example to run load test")
+)
+
+// TestAvailabilityAgainstHighHTTPLoad tests the availability of the proxy with wasm filter against a high HTTP load
+func TestAvailabilityAgainstHighHTTPLoad(t *testing.T) {
+	stdErr, kill := testutil.StartEnvoyWith(*targetExample, t, 8001)
+	defer kill()
+
+	opts := fhttp.HTTPRunnerOptions{}
+	opts.URL = "http://localhost:18000"
+	opts.AllowInitialErrors = true
+	opts.NumThreads = runtime.NumCPU()
+	opts.Percentiles = []float64{99.0}
+	opts.AddAndValidateExtraHeader("Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.t-IDcSemACt8x4iTMCda8Yhe3iZaWbvV5XKSTbuAn0M")
+
+	// Set payload (request body) size
+	fnet.ChangeMaxPayloadSize(*payloadSize * fnet.KILOBYTE)
+	opts.Payload = fnet.Payload
+
+	fortioLog := new(bytes.Buffer)
+	opts.Out = fortioLog
+
+	opts.HTTPReqTimeOut = 5000 * time.Second // Avoid timeouts on huge payloads
+	log.Printf("\tDuration = %d [s], payloadSize = %d [KB]\n", *duration, *payloadSize)
+	opts.QPS = *qps * 1.5 // In order to reach the target QPS, we need to set a little bit higer target QPS.
+	opts.Duration = time.Duration(*duration) * time.Second
+
+	// Run memory profiling to find out memory stability of SDK
+	err := testutil.StartEnvoyMemoryProfile(8001)
+	require.NoError(t, err)
+
+	// Start generating load
+	results, err := fhttp.RunHTTPTest(&opts)
+	require.NoError(t, err)
+
+	// Stop memory profiling
+	memstats, err := testutil.StopEnvoyMemoryProfile()
```

Comment:
what if the test panics before reaching to this line? Looks like the goroutine will leak to me.